src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/invalidjavadocposition/InputInvalidJavadocPosition.java

```diff
@@ -65,4 +65,16 @@ void method() {
         /** violation */ @Deprecated int variable3; // violation
     }
 }
+class GenericConstructor {
+    /** valid */
+    <E extends String> GenericConstructor(E a) {}
+
+    /** valid */
+    private <E extends String> GenericConstructor() {}
+
+    private class InnerClass {
+        /** valid */
+        <E extends String> InnerClass() {}
```

Comment:
please add violation examples

Reply:
Done!

Reply:
@stoyanK7 please add example like this (javadoc on same line)
```
/** javadoc */ <E extends String> InnerClass() {}
```

Reply:
Done!